app/helpers/user_helper.rb

```diff
@@ -0,0 +1,45 @@
+module UserHelper
+  # rubocop:disable Style/GuardClause
+  def friend_button(current_user, user)
+    'You are friends' if friend?(current_user, user)
+  end
+
+  def friend_post(current_user, user)
+    if friend?(current_user, user)
+      render @posts
+    else
+      render @other_posts
+    end
+  end
+
+  def friendship_button(current_user, user)
+    friendship = Friendship.find_by(requestor_id: current_user.id,
+                                    requestee_id: user.id) || Friendship.find_by(requestor_id: user.id,
```

Comment:
There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.